lib/omniauth/strategies/auth0.rb

```diff
@@ -75,16 +77,14 @@ def client
       # Define the parameters used for the /authorize endpoint
       def authorize_params
         params = super
-        params['auth0Client'] = telemetry_encoded
-        parse_query = Rack::Utils.parse_query(request.query_string)
-        params['connection'] = parse_query['connection']
-        params['prompt'] = parse_query['prompt']
+        parsed_query = Rack::Utils.parse_query(request.query_string)
+        params['connection'] = parsed_query['connection']
+        params['prompt'] = parsed_query['prompt']
         params
       end
 
       def build_access_token
-        telemetry_header = { 'Auth0-Client' => telemetry_encoded }
-        options.token_params.merge!(:headers => telemetry_header)
+        options.token_params[:headers] = { 'Auth0-Client' => telemetry_encoded }
```

Comment:
previously this said "merge". Isn't this now a replace rather than a merge? 🤔

Reply:
Should be doing the same. Previous was merging `options.token_params` with `:headers => telemetry_header`, not merging headers (since `:headers` might not be a hash or even exist).